docs/architecture/ADR-002-universal-share-encoding.md

```diff
@@ -0,0 +1,141 @@
+# ADR 009: Universal Share Encoding
+<!-- disable markdownlint MD010 because Go code snippet uses tabs -->
+<!-- markdownlint-disable MD010 -->
+
+## Terminology
+
+- **nid** (8 bytes): namespace id
+- **reserved** (1 byte): is the location of the first transaction, ISR, or evidence in this share if there is one and `0` if there isn't one
+- **message length** (varint 1 to 10 bytes): is the length of the entire message in bytes
+- **compact share**: a type of share that can accomodate multiple units. Currently, compact shares are used for transactions, ISRs, and evidence to efficiently pack this information into as few shares as possible.
+- **sparse share**: a type of share that can accomodate zero or one unit. Currently, sparse shares are used for messages.
+
+## Context
+
+The current compact share format is:<br>`nid (8 bytes) | reserved (1 byte) | data`
```

Comment:
This statement isn't congruent with [this implementation](https://github.com/celestiaorg/celestia-app/blob/main/pkg/shares/parse_contiguous_shares.go#L49). In particular, it looks like data length is encoded as a varint into compact shares.

```suggestion
The current compact share format is:

- First share of namespace: <br>`nid (8 bytes) | reserved (1 byte) | message length (varint 1 to 10 bytes) | data`
- Contiguous share in namespace:<br>`nid (8 bytes) | reserved (1 byte) | data`
```